tests/test_discretisation/test_equal_frequency_discretiser.py

```diff
@@ -58,13 +58,37 @@ def test_error_if_input_df_contains_na_in_fit(df_na):
 
 def test_error_if_input_df_contains_na_in_transform(df_vartypes, df_na):
     # test case 4: when dataset contains na, transform method
-    with pytest.raises(ValueError):
-        transformer = EqualFrequencyDiscretiser()
+    msg = "During the discretisation, NaN values were introduced " \
```

Comment:
we need a new test to test your new functionality. But we also need the old test, because that is testing something else.

Could you please restore the old test and pack your new code into a new test?